src/Umbraco.Web/Editors/MemberController.cs

```diff
@@ -186,53 +191,31 @@ public MemberDisplay GetEmpty(string contentTypeAlias = null)
                 throw new HttpResponseException(Request.CreateValidationErrorResponse(forDisplay));
             }
 
-            // TODO: WE need to support this! - requires UI updates, etc...
-            if (_provider.RequiresQuestionAndAnswer)
-            {
-                throw new NotSupportedException("Currently the member editor does not support providers that have RequiresQuestionAndAnswer specified");
-            }
-
             //We're gonna look up the current roles now because the below code can cause
             // events to be raised and developers could be manually adding roles to members in
             // their handlers. If we don't look this up now there's a chance we'll just end up
             // removing the roles they've assigned.
-            var currRoles = Roles.GetRolesForUser(contentItem.PersistedContent.Username);
+            var currRoles = Services.MemberService.GetAllRoles(contentItem.PersistedContent.Username);
             //find the ones to remove and remove them
             var rolesToRemove = currRoles.Except(contentItem.Groups).ToArray();
 
-            string generatedPassword = null;
             //Depending on the action we need to first do a create or update using the membership provider
             // this ensures that passwords are formatted correctly and also performs the validation on the provider itself.
             switch (contentItem.Action)
             {
                 case ContentSaveAction.Save:
-                    generatedPassword = UpdateWithMembershipProvider(contentItem);
+                    UpdateMemberData(contentItem);
                     break;
                 case ContentSaveAction.SaveNew:
-                    MembershipCreateStatus status;
-                    CreateWithMembershipProvider(contentItem, out status);
-
-                    // save the ID of the creator
-                    contentItem.PersistedContent.CreatorId = Security.CurrentUser.Id;
+                    contentItem.PersistedContent = CreateMemberData(contentItem);
```

Comment:
There is something wrong going on here.. Before this `contentItem.PersistedContent.RawPasswordValue` contains the value inputted in the UI. After this line the value is empty.

The result is also that the password i Use when I create the member, cannot be used to sign in the member

Reply:
Thanks and good catch, have fixed this up and removed more membership provider stuff that I overlooked.